uiuc_transportation_demand/functions.py

```diff
@@ -0,0 +1,282 @@
+        GGECost.update({fuel: CostDictionary[fuel] / GGEDictionary[fuel]})
+
+    return GGECost
+
+
+def fuel_equivalent(Fuel):
+
+    """
+    This function returns the total amount of desired fuel type
+    required to fully manage the UIUC fleet in 2019 assuming it
+    was readily available based on the fuel_data.csv
+
+    Parameters:
+    -----------
+    Fuel: string
+        Desired fuel type to analyze ('Gasoline', 'Diesel', 'E85',
+        'Hydrogen', or 'Electicity')
+
+    Returns:
+    --------
+    amount: float
+        The total amount of units for desired fuel technology that would
+        have powered the UIUC fleet in 2019
+
+    """
+
+    if type(Fuel) != str:
+        raise TypeError(
+         'Please ensure that the input parameter is of type "str".'
+        )
+
+    Dict = gge_dictionary()
+
+    if Fuel not in Dict:
+        raise IndexError(
+         f'{Fuel} not supported. Try "Gasoline", "Diesel", "E85", "Hydrogen", or "Electricity".'
+        )
+
+    FuelData = 'fuel_data.csv'
+
+    with open(FuelData, 'r') as i:
+        fueldata = i.readlines()
+
+    lines = list()
+
+    for line in fueldata:
+        lines.append(line.split(','))
+
+    counter = 0
+
+    for line in lines:
+        if line[6] == 'Asset':
+            continue
+
+        else:
+            if line[16].split(' ')[2] == 'UNLEADED':
+
+                counter += Dict['Gasoline'] * (float(line[16].split(' ')[0]))
+
+            elif line[16].split(' ')[2] == 'DIESEL':
+
+                counter += Dict['Diesel'] * (float(line[16].split(' ')[0]))
+
+            elif line[16].split(' ')[2] == 'E85':
+
+                counter += Dict['E85'] * (float(line[16].split(' ')[0]))
+
+    allowed_fuels_function = {
+        'Gasoline': round(counter, 2),
+        'Diesel': round(counter / Dict['Diesel'], 2),
+        'E85': round(counter / Dict['E85'], 2),
+        'Hydrogen': round(counter / Dict['Hydrogen']),
+        'Electricity': round(counter / Dict['Electricity'])
+    }
+
+    amount = allowed_fuels_function[Fuel]
+
+    return amount
+
+
+def fuel_equivalent_cost(Fuel):
+
+    """
+    This function returns the total cost of operating the 2019 UIUC
+    fleet on the desired fuel technology alone.
+
+    Parameters:
+    -----------
+    Fuel: string
+        Type of fuel technology to be analyzed ('Gasoline', 'Diesel',
+        'E85', 'Hydrogen', or 'Electricity').
+
+    Returns:
+    --------
+    cost: float
+        The total cost in $ of using only that fuel technology type
+        to replace the entire 2019 UIUC fleet fuel.
+
+    """
+
+    if type(Fuel) != str:
+        raise TypeError(
+         'Please ensure that the input parameter is of type "str".'
+        )
+
+    CostDictionary = unit_cost_dictionary()
+
+    GGEDictionary = gge_dictionary()
+
+    if Fuel not in GGEDictionary:
+        raise IndexError(
+         f"{Fuel} not supported. Try 'Gasoline', 'Diesel', 'E85', 'Hydrogen', or 'Electricity'."
```

Comment:
```suggestion
         f"{Fuel} not supported. Try {list(GGEDictionary.keys())}."
```
This accomplishes the same thing, but with fewer characters.

Reply:
Ah, I just saw this suggestion after I altered the file, I will make these final changes.